### PR TITLE
Feature/OP-754: Create a Link

### PR DESCRIPTION
**Pull request by gzhou1**
Created Frontend form to add a link to a request.
Implemented Backend functions to generate event entries and sends email notifications to the requester and bccs all agency users.

### CODE REVIEW

File: config.py [hunk 1]
```diff
@@ -70,6 +71,7 @@ class DevelopmentConfig(Config):
     SQLALCHEMY_DATABASE_URI = (os.environ.get('DATABASE_URL') or
                                'postgresql://localhost:5432/openrecords_v2_0_dev')
     # Using Vagrant? Try: 'postgresql://vagrant@/openrecords_v2_0_dev'
+    ELASTICSEARCH_ENABLED = False
```
Comment (Palisand) [inline config.py]: This should be set in .env

File: app/response/utils.py [hunk 2]
```diff
@@ -47,15 +48,15 @@
 
 def add_file(request_id, filename, title, privacy):
     """
-    Creates and stores the file object for the specified request.
+    Creates and store the file object for the specified request.
```
Comment (Palisand) [inline app/response/utils.py]: why was this changed?

File: app/response/utils.py [hunk 3]
```diff
@@ -47,15 +48,15 @@
...
     Gets the file mimetype from a helper function in lib.file_utils
 
     :param request_id: Request ID that the file is being added to
     :param filename: The secured_filename of the file.
     :param title: The title of the file which is entered by the uploader.
     :param privacy: The privacy option of the file.
 
-    :return: Stores the file metadata into the Files table.
+    :return: Store the file metadata into the Files table.
```
Comment (Palisand) [inline app/response/utils.py]: This function does not return anything.

**Comment by joelbcastillo** [comment]
Generally, for comments, a description of what the function does should be before the parameter information. First line of the docstring should be a simple description of the full function. After that, the description can be more detailed.

File: app/response/utils.py (hunk 2 above)
Comment (joelbcastillo) [inline app/response/utils.py]: I would suggest following [PEP-257](https://www.python.org/dev/peps/pep-0257)

File: config.py (hunk 1 above)
Comment (joelbcastillo) [inline config.py]: Replace this with: `ELASTICSEARCH_ENABLED = eval(str(os.environ.get('ELASTICSEARCH_ENABLED')))`

File: config.py [hunk 4]
```diff
@@ -71,6 +71,8 @@ class DevelopmentConfig(Config):
...
+    MAGIC_FILE = ''
```
Comment (joelbcastillo) [inline config.py]: MAGIC_FILE = eval(str(os.environ.get('MAGIC_FILE'))) or ''

File: app/response/utils.py (hunk 2 above)
Comment (gzhou1) [inline app/response/utils.py]: Updated docstring to follow PEP-257

File: app/response/utils.py [hunk 5]
```diff
@@ -305,21 +303,20 @@ def process_email_template_request(request_id, data):
     """
     page = flask_request.host_url.strip('/') + url_for('request.view', request_id=request_id)
     agency_name = Requests.query.filter_by(id=request_id).first().agency.name
+    email_template = os.path.join(current_app.config['EMAIL_TEMPLATE_DIR'], data['template_name'])
     # process email template for extension
     if data['type'] == 'extension_email':
+        # email_template = os.path.join(current_app.config['EMAIL_TEMPLATE_DIR'], data['template_name'])
```
Comment (gzhou1) [inline app/response/utils.py]: Removed

File: app/response/utils.py [hunk 6]
```diff
@@ -330,11 +327,25 @@ def process_email_template_request(request_id, data):
             if file_['privacy'] != PRIVATE:
                 filename = file_['filename']
                 files_links[filename] = "http://127.0.0.1:5000/request/view/{}".format(filename)
+        email_template = os.path.join(current_app.config['EMAIL_TEMPLATE_DIR'], data['template_name'])
```
Comment (gzhou1) [inline app/response/utils.py]: Removed

File: config.py (hunk 1 above)
Comment (gzhou1) [inline config.py]: Replaced

File: config.py (hunk 4 above)
Comment (gzhou1) [inline config.py]: Replaced
